commands/token.go

```diff
@@ -0,0 +1,134 @@
+	Example: "  generate testuser test-token",
+	RunE:    withClient(generateTokenForAUserCmdF),
+	Args:    cobra.ExactArgs(2),
+}
+
+var RevokeUserTokenCmd = &cobra.Command{
+	Use:     "revoke [user] [token-ids]",
+	Short:   "Revoke tokens for a user",
+	Long:    "Revoke tokens for a user",
+	Example: "  revoke testuser test-token-id",
+	RunE:    withClient(revokeTokenForAUserCmdF),
+	Args:    cobra.MinimumNArgs(2),
+}
+
+var ListUserTokensCmd = &cobra.Command{
+	Use:     "list [user]",
+	Short:   "List users tokens",
+	Long:    "List the tokens of a user",
+	Example: "  user tokens testuser",
+	RunE:    withClient(listTokensOfAUserCmdF),
+	Args:    cobra.ExactArgs(1),
+}
+
+func init() {
+	ListUserTokensCmd.Flags().Int("page", 0, "Page number to fetch for the list of users")
+	ListUserTokensCmd.Flags().Int("per-page", 200, "Number of users to be fetched")
+	ListUserTokensCmd.Flags().Bool("all", false, "Fetch all tokens. --page flag will be ignore if provided")
+	ListUserTokensCmd.Flags().Bool("active", true, "List only active tokens")
+	ListUserTokensCmd.Flags().Bool("inactive", false, "List only inactive tokens")
+
+	TokenCmd.AddCommand(
+		GenerateUserTokenCmd,
+		RevokeUserTokenCmd,
+		ListUserTokensCmd,
+	)
+
+	RootCmd.AddCommand(
+		TokenCmd,
+	)
+}
+
+func generateTokenForAUserCmdF(c client.Client, command *cobra.Command, args []string) error {
+	userArg := args[0]
+	user := getUserFromUserArg(c, userArg)
+	if user == nil {
+		return errors.Errorf("could not retrieve user information of %q", userArg)
+	}
+
+	token, res := c.CreateUserAccessToken(user.Id, args[1])
+	if res.Error != nil {
+		return errors.Errorf("could not create token for %q: %s", userArg, res.Error.Error())
+	}
+	printer.PrintT("{{.Token}}: {{.Description}}", token)
+
+	return nil
+}
+
+func listTokensOfAUserCmdF(c client.Client, command *cobra.Command, args []string) error {
+	page, _ := command.Flags().GetInt("page")
+	perPage, _ := command.Flags().GetInt("per-page")
+	showAll, _ := command.Flags().GetBool("all")
+	active, _ := command.Flags().GetBool("active")
+	inactive, _ := command.Flags().GetBool("inactive")
+
+	if showAll {
+		page = 0
+	}
+
+	userArg := args[0]
+
+	user := getUserFromUserArg(c, userArg)
+	if user == nil {
+		return errors.Errorf("could not retrieve user information of %q", userArg)
+	}
+
+	tokens, res := c.GetUserAccessTokensForUser(user.Id, page, perPage)
+	if res.Error != nil {
+		return errors.Errorf("could not retrieve tokens for user %q: %s", userArg, res.Error.Error())
+	}
+	if len(tokens) == 0 {
+		return errors.Errorf("there are no tokens for the %q", userArg)
+	}
+
+	for _, t := range tokens {
+		if t.IsActive && !inactive {
+			printer.PrintT("{{.Id}}: {{.Description}}", t)
+		}
+		if !t.IsActive && !active {
+			printer.PrintT("{{.Id}}: {{.Description}}", t)
+		}
+	}
+	return nil
+}
+
+func revokeTokenForAUserCmdF(c client.Client, command *cobra.Command, args []string) error {
+	userArg := args[0]
+	user := getUserFromUserArg(c, userArg)
```

Comment:
is the user required in revoking a token? doesn't look like we're doing anything with it though.

Reply:
Good point @ali-farooq0, my idea is to retain consistency with token creation but was not so sure though. I can remove it 👍